downloads/swan-lake-release-notes/swan-lake-2201.6.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,353 @@
+    }
+    ```
+
+- A bug that allowed assigning nil to a record field with member access expressions when there are no fields of optional types has been fixed. This previously resulted in a runtime panic if the value was nil.
+
+    ```ballerina
+    type Employee record {|
+        string id;
+        string name;
+    |};
+
+    public function main() {
+        Employee employee = {
+            name: "Jo",
+            id: "E12321"
+        };
+
+        string key = "name";
+        employee[key] = (); // Compilation error now.
+
+        map<string> data = {
+            name: "Jo Doe",
+            dept: "IT"
+        };
+
+        foreach string mkey in employee.keys() {
+            // `data[key]` will be nil if the key is not present in `data`.
+            // E.g., `data[key]` is nil when `key` is `name`.
+            employee[mkey] = data[mkey]; // Compilation error now.
+        }
+    }
+    ```
+    
+- Fixed a bug with the XML parser error messages that showed dependency information. The error message prefix `failed to create xml` is now changed to `failed to parse xml` to have a single prefix for all the XML-parsing error messages.
+
+    For example, consider the content below of the `invalid_xml.txt` file.
+
+    ```
+    <!-- comments cannot have -- in it -->
+    ```
+
+    When the following Ballerina code is run,
+
+    ```ballerina
+    import ballerina/io;
+
+    public function main() returns error? {
+        xml readResult = check io:fileReadXml("invalid_xml.txt");
+    }
+    ```
+
+    it gives the error below.
+
+    ```
+    error: failed to parse xml: String '--' not allowed in comment (missing '>'?)
+     at [row,col {unknown-source}]: [1,4]
+    ```
+
+- The `Environment`, `Future`, and `Runtime` classes in the `io.ballerina.runtime.api` package are refactored to abstract classes. Creating an instance of those classes is incorrect.
+
+- A typedesc value (returned by evaluating the `typeof` expression on a result of the `value:cloneWithType` and `value:fromJsonWithType` functions) is changed to give the correct type-reference type.
+
+    ```ballerina
+    import ballerina/io;
+
+    type IntArray int[];
+
+    public function main() returns error? {
+        float[] arr = [1.0, 2.0];
+        IntArray result = check arr.cloneWithType();
+        io:println(typeof result); // Prints 'typedesc IntArray'.
+    }
+    ```
+
+## Language updates
+
+### Bug fixes
+
+To view other bug fixes, see the [GitHub milestone for Swan Lake 2201.6.0](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+label%3ATeam%2FCompilerFE+milestone%3A2201.6.0+is%3Aclosed+label%3AType%2FBug).
+
+## Runtime updates
+
+### New features
+
+#### New Runtime Java APIs
+
+Introduced the `getInitMethod()` API in the `io.ballerina.runtime.api.types.ObjectType` class to get the method type of the initializer method of Ballerina objects.
+
+```java
+MethodType getInitMethod();
+```
+
+### Deprecations
```

Comment:
Be consistent on the order of the sub titles. Here `Deprecations` comes after `New features` while in standard library `Deprecations` comes first. IMO, `Deprecations` should come first.

Reply:
Fixed.